dojo_plugin/utils/awards.py

```diff
@@ -74,13 +60,30 @@ def get_belts():
     return result
 
 def update_awards(user):
-    current_belts = get_user_belts(user)
-    for belt in current_belts:
-        belt_award = Belts.query.filter_by(user=user, name=belt).first()
-        if belt_award:
+    current_belts = [belt.name for belt in Belts.query.filter_by(user=user)]
+    for belt, dojo_id in BELT_REQUIREMENTS.items():
+        if belt in current_belts:
+            continue
+        dojo = Dojos.query.filter(Dojos.official, Dojos.id == dojo_id).first()
+        if not (dojo and dojo.completed(user)):
             continue
```

Comment:
I think the remaining issue is that, due to this continue, belts are no longer cumulative. Shouldn't this be a `break`?

Reply:
Good call